src/components/LmsConfigurations/index.jsx

```diff
@@ -33,32 +36,52 @@ class LmsConfigurations extends React.Component {
       LmsApiService.fetchDegreedConfig(this.props.enterpriseId),
       LmsApiService.fetchCornerstoneConfig(this.props.enterpriseId),
     ]).then((responses) => {
-      if (responses.some(response => response.reason?.request.status === 400
-          || response.reason?.request.status > 404)) {
+      if (
+        responses.some(
+          (response) => response.reason?.request.status === 400
+            || response.reason?.request.status > 404,
+        )
+      ) {
         const errorRsp = responses.filter(
-          response => response.reason?.request.status !== 404 && response.status === 'rejected',
+          (response) => response.reason?.request.status !== 404
+            && response.status === 'rejected',
         );
         const errorMsgs = [];
         let status = '';
         errorRsp.forEach((error) => {
           errorMsgs.push(error.reason.request.statusText);
           status = error.reason.request.status;
         });
-        this.setState({ error: { response: { status }, message: errorMsgs.join(';') }, loading: false });
+        this.setState({
+          error: { response: { status }, message: errorMsgs.join(';') },
+          loading: false,
+        });
       } else {
         this.setState({
-          moodleConfig: responses[0].status === 'fulfilled'
-            ? responses[0].value.data.results[0] : null,
-          canvasConfig: responses[1].status === 'fulfilled'
-            ? responses[1].value.data.results[0] : null,
-          blackboardConfig: responses[2].status === 'fulfilled'
-            ? responses[2].value.data.results[0] : null,
-          sapsfConfig: responses[3].status === 'fulfilled'
-            ? responses[3].value.data.results[0] : null,
-          degreedConfig: responses[4].status === 'fulfilled'
-            ? responses[4].value.data.results[0] : null,
-          cornerstoneConfig: responses[5].status === 'fulfilled'
-            ? responses[5].value.data.results[0] : null,
+          moodleConfig:
+            responses[0].status === 'fulfilled'
+              ? responses[0].value.data.results[0]
+              : null,
+          canvasConfig:
+            responses[1].status === 'fulfilled'
+              ? responses[1].value.data.results[0]
+              : null,
+          blackboardConfig:
+            responses[2].status === 'fulfilled'
+              ? responses[2].value.data.results[0]
+              : null,
+          sapsfConfig:
+            responses[3].status === 'fulfilled'
+              ? responses[3].value.data.results[0]
+              : null,
+          degreedConfig:
+            responses[4].status === 'fulfilled'
+              ? responses[4].value.data.results[0]
+              : null,
+          cornerstoneConfig:
+            responses[5].status === 'fulfilled'
+              ? responses[5].value.data.results[0]
+              : null,
```

Comment:
This is repeated piece of code. We can use iteration for this. Even write a helper function
```suggestion
     ...mapPromiseSettlingResults(results, ({ data }) => data.results[0], [
        'moodleConfig',
        'canvasConfig',
        'blackboardConfig',
        'sapsfConfig',
        'degreedConfig',
        'cornerstoneConfig',
     ])
```

Reply:
```javascript
function mapPromiseSettlingResults(results, getValue, orderedKeys) {
  const mapped = {};
  orderedKeys.forEach((key, i) => {
     const { status, value } = results[i];
     mapped[key] = status === 'fulfilled' ? getValue(value) : null;
  });
  return mapped;
}
```